### ISSUE TITLE
Signal + noise composition interface

### ISSUE THREAD

**Issue by msyriac** [simonsobs/mapsims#31]
What is the recommended code interface to make a joint signal + noise sim?
I would strongly advocate that the default noise sim is *not beam deconvolved*. Rather, a composite sim of signal and noise should look like:
beam(signal) + noise.
This makes the result look like what is in a real data map, and makes the noise map well behaved.

So, it would be good to have a signal + noise composition interface that does the following:

bandpass_integrate(beam(cmb + galactic + extragalactic)) + noise

e.g.
```
sgen = SOSim()
sgen.get_sim(band,bandpass_integrate=True,beam=True,include_noise=True,include_galactic=True,include_extragalactic=True)
```

This probably exists somewhere already?

**Comment by ajvanengelen** [comment]
Agreed - as you know that is the approach we have taken for the ACT sims.
________________________________

**Comment by zonca** [comment]
this is what `MapSim` does: https://mapsims.readthedocs.io/en/latest/api/mapsims.MapSim.html#mapsims.MapSim

it can start from a configuration file or from classes, and also has a command line tool, see https://mapsims.readthedocs.io/en/latest/usage.html

**Comment by msyriac** [comment]
Couple of things I see here:
1. We need to add CAR support
2. It doesn't look like beam convolution is ever done anywhere, instead the noise needs to be obtained with "beam_correction=True"

Is (2) correct?

**Comment by zonca** [comment]
We ran the last set of noise simulations with no beam correction, see https://github.com/simonsobs/map_based_simulations/blob/master/201906_noise_no_lowell/noise.toml

The next set should instead have it included? https://github.com/simonsobs/map_based_simulations/issues/21

**Comment by msyriac** [comment]
I think it always makes sense to save or return noise simulations with no beam deconvolution. And similarly signal sims should always be beam convolved. This makes the outputs well behaved / sane. The user can then simply treat:
signal + noise
as a representation of the kind of map SO mapmakers will provide us.

**Comment by zonca** [comment]
so we will run also the new sims with `beam_correction=False`?

**Comment by msyriac** [comment]
Yes, but are you also separately running and saving signal sims? Will you apply beam convolution to those?

**Comment by zonca** [comment]
this release will be only noise, but previous and next signal releases all have beam convolution.

**Comment by zonca** [comment]
@msyriac @ajvanengelen here is how it works now:

https://gist.github.com/zonca/5cc4e2d27cbd8422f16e1af3e5e29bfd

It would be nicer to have `sim.execute(seed)` but I wouldn't want to make too many changes to the pipeline. So I think this is easy enough to use, suggestions?

**Comment by zonca** [comment]
After this release I'll work on the foreground simulations again, I think at that point it will be easier to test maps+noise simulations and work on improving the interface. I would like to check if I can initialize all the foreground and noise classes without providing NSIDE, channel, and num, then call:

    sim.execute(channel="ST0", nside=64, num=10)
